tock-responder: Initial commit of Tock responder support

alistair23 opened the pull request:
Add initial support for a Tock SPDM responder. This is an application
that can run on Tock and handle SPDM MCTP packets.

Currently this is in very early stages. The application runs, but does
not yet act as a responder.

twilfredo commented:
There's a typo in the commit msg after my SoB. Also, libspdm has the mctp encode and decode functions. Similar to the DOE ones, I wonder if we could just use those instead, that way libspdm handles it all.

alistair23 commented:
Fixed the typo!

I have removed libmctp. It might end up being simpler to use the Rust library though, instead of libspdm's functions. It might end up using less memory as well. We can always add it back if required
